### PR TITLE
protocgen, readme: add additional setup instructions

### PR DESCRIPTION
**Pull request by matildepark**
Just writing down some documentation for stuff I was getting stuck on while setting this up on my M1 Mac. Some of this (like that `$GOPATH` is there at all) is expected for developers to have [based on the Tendermint docs](https://docs.tendermint.com/master/app-dev/getting-started.html#install), but instructions for a full-stack product like this are best made thorough and explicit, I think.

Secondly, the protocgen script assumes the user's environment, so this adds some conditional checks so Mac users can build without hassle.

Let me know if any of this is off base...

### CODE REVIEW

**Comment by vlamitin** [comment]
Thanks for pointing this out!
This is in a very deep develpment now, and currently used latest cosmos-sdk node is now yet works correctly with this front-end. And that's why only development in linux was supported.

File: README.md [hunk 1]
```diff
@@ -3,8 +3,15 @@
 ## Prerequisites
 - `node 12.*`
 - `go 1.17.*`, `unzip`, `curl` (for proto generation)
+- [Keplr Chrome extension](https://www.keplr.app/)
+- `tendermint`
```
Comment (vlamitin) [inline README.md:7]: no tendermint install is needed, just go to build cosmos-sdk simd

File: README.md [hunk 2]
```diff
@@ -3,8 +3,15 @@
...
 
 ## Usage
+
+Ensure you have a Tendermint node running locally and that CORS is allowed from any origin in `config.toml`.
```
Comment (vlamitin) [inline README.md:11]: In latest cosmos-sdk cors directives in config.toml is ignored (we are communicating with cosmos-sdk development team on it), so workaround for development for now is some chrome extension to bypass this, like this one https://chrome.google.com/webstore/detail/digfbfaphojjndkpccljibejjbppifbc (you can point it here in README)

**Review by vlamitin — CHANGES_REQUESTED** [review]

**Comment by matildepark** [comment]
Closing because mostly remedied by new README.

**Pull request closed without merging**